src/renderer/js/redux/reducers/app.js

```diff
@@ -1,22 +1,45 @@
-import * as types from "constants/action_types";
-import * as modalTypes from "constants/modal_types";
+// @flow
 
+import * as types from "../../constants/action_types";
+import * as modalTypes from "../../constants/modal_types";
+
+// $FlowIssue: Flow cannot find election
 const { remote } = require("electron");
+
 const application = remote.app;
 const win = remote.BrowserWindow.getFocusedWindow();
 
 const reducers = {};
-const defaultState = {
+type appState = {
+  isLoaded: boolean,
+  modal: mixed,
+  modalProps: mixed,
+  platform: string,
+  upgradeSkipped: boolean,
+  daemonVersionMatched: ?boolean,
+  daemonReady: boolean,
+  hasSignature: boolean,
+  badgeNumber: number,
+  volume: number,
+  downloadProgress?: number,
+  upgradeDownloading?: boolean,
```

Comment:
Any particular reason are these optional properties that we don't set on `defaultState`? IMO optional properties should generally be avoided.

Reply:
optional properties can be thought of as implicitly there, but set to "undefined". I am more than happy to add them, though.

The one edge case I could see, and I will double check this, is, as I was saying in slack:

downloadProgress?: number could be a number or undefined
downloadProgress: ?number could be a number or undefined or null

In order to remove optional fields, we would have to either allow null or determine a non-null default value for each field.